controllers/managedocs_controller.go

```diff
@@ -2045,3 +2096,28 @@ func (r *ManagedOCSReconciler) setDeviceSetCount(deviceSet *ocsv1.StorageDeviceS
 		deviceSet.Count = currDeviceSetCount
 	}
 }
+
+func (r *ManagedOCSReconciler) reconcileODFSubscription() error {
+	subList := &opv1a1.SubscriptionList{}
+	if err := r.list(subList); err != nil {
+		return fmt.Errorf("Unable to list subscriptions %v", err)
+	}
+	actualSub := &opv1a1.Subscription{}
+	for index := range subList.Items {
+		item := subList.Items[index]
+		if item.Spec.Package == odfPackageName {
+			actualSub = &item
```

Comment:
- rest of the function is assuming that you actually found the subscription 🤔 
- in failure case, `r.update` fails incorrectly

Reply:
Updated